Update CrowdStrikeFalcon.py

UlisseDallara opened the pull request:
Changed 'get_behavior_command' command as it would not pull results.

<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ x] No

## Must have
- [ ] Tests
- [ ] Documentation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/demisto/content?pullRequest=24320) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=24320) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=24320) it.</sub>

yaakovpraisler reviewed (changes requested):
Hi @UlisseDallara, Thank you for your contribution!
It was a little difficult for me to understand the change you made here, Can you please explain what you did here and why?
Also, when I checked your change in my XSOAR env I've got a `Bad Request` error, did it work for you?

Also, for pushing your changes into our content we'll need to bump the pack version and add release notes
https://xsoar.pan.dev/docs/documentation/release-notes

Please feel free to reach out to me with any questions - I'm available here or on slack :)
Thanks again

UlisseDallara commented:
Hi Yaakov,

Thanks for getting back to me.

Basically the automation: 'cs-falcon-get-behavior' does not work currently.

If you have visibility of the support ticket 02445526 I put more details there, but I liaised with Crowdstrike support and they have confirmed the behaviour IDs are correct, they also mentioned that the endpoint that need to be queried is the one I used, rather than the one currently in use.

Yes I ran the automations many times and it works as expected.

I bumped out the version, I am checking where to add the release notes, (it is the first time I am pushing content to you).

Many thanks,

Regards,

Ulisse
________________________________
From: Yaakov Praisler ***@***.***>
Sent: Tuesday, February 7, 2023 12:12 PM
To: demisto/content ***@***.***>
Cc: Ulisse Dallara ***@***.***>; Mention ***@***.***>
Subject: Re: [demisto/content] Update CrowdStrikeFalcon.py (PR #24320)


@yaakovpraisler requested changes on this pull request.

Hi @UlisseDallara<https://github.com/UlisseDallara>, Thank you for your contribution!
It was a little difficult for me to understand the change you made here, Can you please explain what you did here and why?
Also, when I checked your change in my XSOAR env I've got a Bad Request error, did it work for you?

Also, for pushing your changes into our content we'll need to bump the pack version and add release notes
https://xsoar.pan.dev/docs/documentation/release-notes

Please feel free to reach out to me with any questions - I'm available here or on slack :)
Thanks again

—
Reply to this email directly, view it on GitHub<https://github.com/demisto/content/pull/24320#pullrequestreview-1286980006>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A5UZLHM6AXKCTHIPHYS7DE3WWI32FANCNFSM6AAAAAAUQFP5SM>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

yaakovpraisler commented:
Hi Ulisse,

It sounds a bit strange to me, because for me the `cs-falcon-get-behavior` command works fine before your fix and I've got the error after updating your fix.

Are you sure you don't have other custom changes in the integration that might cause the issue? Can you please check with the currently updated pack version in the marketplace if the automation issue still happens?

I don't have visibility of the support ticket, can you please send me the details? or refer your contact CS person to me with our internal ticket?

I also see that the current endpoint is for detections and your endpoint is for incidents, so maybe they are two different things? if so, you can add a separate command for getting incident behavior.

Thanks again,

Yaakov

UlisseDallara commented:
Hi Yaakow,

My query is indeed for incidents, but I do not see mentioned anywhere in the integration itself or its documentation that the current solution is meant for detections only.

OK, I can add an automation named 'cs-falcon-get-incident-behavior' then.

But if I run "!cs-falcon-get-detections-for-incident incident_id=*" I get a table with behaviors, detections, incidents, but none of those behavior_id works?

Which ID are using? We have a use case for detections too, so I will need that to work as well.

The incident is with Sakshi Rai.

Thanks,

Regards,

Ulisse
________________________________
From: Yaakov Praisler ***@***.***>
Sent: Tuesday, February 7, 2023 2:00 PM
To: demisto/content ***@***.***>
Cc: Ulisse Dallara ***@***.***>; Mention ***@***.***>
Subject: Re: [demisto/content] Update CrowdStrikeFalcon.py (PR #24320)


Hi Ulisse,

It sounds a bit strange to me, because for me the cs-falcon-get-behavior command works fine before your fix and I've got the error after updating your fix.

Are you sure you don't have other custom changes in the integration that might cause the issue? Can you please check with the currently updated pack version in the marketplace if the automation issue still happens?

I don't have visibility of the support ticket, can you please send me the details? or refer your contact CS person to me with our internal ticket?

I also see that the current endpoint is for detections and your endpoint is for incidents, so maybe they are two different things? if so, you can add a separate command for getting incident behavior.

Thanks again,

Yaakov

—
Reply to this email directly, view it on GitHub<https://github.com/demisto/content/pull/24320#issuecomment-1420822123>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A5UZLHJSYZK3BVDDQDFFBALWWJIRJANCNFSM6AAAAAAUQFP5SM>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

UlisseDallara commented:
Hi Yaakow,

As discussed I reverted previous change and added an automation: "cs-falcon-get-incident-behavior".

I can confirm that the native command 'cs-falcon-get-behavior' gives me a 'no entried' results, while mine works.

Please let me know if anything else is needed,

Many thanks,

Regards,

Ulisse      
________________________________
From: Ulisse Dallara ***@***.***>
Sent: Tuesday, February 7, 2023 2:39 PM
To: demisto/content ***@***.***>; demisto/content ***@***.***>
Cc: Mention ***@***.***>
Subject: Re: [demisto/content] Update CrowdStrikeFalcon.py (PR #24320)

Hi Yaakow,

My query is indeed for incidents, but I do not see mentioned anywhere in the integration itself or its documentation that the current solution is meant for detections only.

OK, I can add an automation named 'cs-falcon-get-incident-behavior' then.

But if I run "!cs-falcon-get-detections-for-incident incident_id=*" I get a table with behaviors, detections, incidents, but none of those behavior_id works?

Which ID are using? We have a use case for detections too, so I will need that to work as well.

The incident is with Sakshi Rai.

Thanks,

Regards,

Ulisse
________________________________
From: Yaakov Praisler ***@***.***>
Sent: Tuesday, February 7, 2023 2:00 PM
To: demisto/content ***@***.***>
Cc: Ulisse Dallara ***@***.***>; Mention ***@***.***>
Subject: Re: [demisto/content] Update CrowdStrikeFalcon.py (PR #24320)


Hi Ulisse,

It sounds a bit strange to me, because for me the cs-falcon-get-behavior command works fine before your fix and I've got the error after updating your fix.

Are you sure you don't have other custom changes in the integration that might cause the issue? Can you please check with the currently updated pack version in the marketplace if the automation issue still happens?

I don't have visibility of the support ticket, can you please send me the details? or refer your contact CS person to me with our internal ticket?

I also see that the current endpoint is for detections and your endpoint is for incidents, so maybe they are two different things? if so, you can add a separate command for getting incident behavior.

Thanks again,

Yaakov

—
Reply to this email directly, view it on GitHub<https://github.com/demisto/content/pull/24320#issuecomment-1420822123>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A5UZLHJSYZK3BVDDQDFFBALWWJIRJANCNFSM6AAAAAAUQFP5SM>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

yaakovpraisler commented:
Hi Ulisse,
You're right that the command s not documented that it's for detections, if you can please add it to the command description in the yml file.

Also, we need to add your new command to more places like to the yml and README files and it need also a unit-test and a task in the test-playbook.

You can find the documentation for all of these [here](https://xsoar.pan.dev/docs/contributing/checklist) and if you're missing something or you need help feel free to reach me out here or on Slack DFIR.

UlisseDallara commented:
Hi Yaakov,

I edited the yaml. I would not be able to share output from the API as it would include information that cannot be shared due to their nature.

I do not see the test playbook?

Many thanks,

Regards,

Ulisse

yaakovpraisler commented:
Hi Ulisse,

I tryied to test the command myself but I couldn't find a correct value for the incident behavior id argument, from which command I can get this?

The test-playbook exist under the TestPlaybooks dir in the integration, and as I mentioned above you need also document the command in the README file an add a unit test.

If you have more questions or/and you see that you are not getting along with that, then we can schedule a call and discuss it.

Thanks.

UlisseDallara commented:
Hi Yaakov,

That would be the output of: 'cs-falcon-get-detections-for-incident'.

Are you free on Thursday for a call? I am available all day 8 to 4.30 GMT time except 2.30-3.

Thanks,

Regards,

Ulisse
________________________________
From: Yaakov Praisler ***@***.***>
Sent: Tuesday, February 14, 2023 9:41 AM
To: demisto/content ***@***.***>
Cc: Ulisse Dallara ***@***.***>; Mention ***@***.***>
Subject: Re: [demisto/content] Update CrowdStrikeFalcon.py (PR #24320)


Hi Ulisse,

I tryied to test the command myself but I couldn't find a correct value for the incident behavior id argument, from which command I can get this?

The test-playbook exist under the TestPlaybooks dir in the integration, and as I mentioned above you need also document the command in the README file an add a unit test.

If you have more questions or/and you see that you are not getting along with that, then we can schedule a call and discuss it.

Thanks.

—
Reply to this email directly, view it on GitHub<https://github.com/demisto/content/pull/24320#issuecomment-1429422941>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A5UZLHORWTGVLUT6DLMC72DWXNHMVANCNFSM6AAAAAAUQFP5SM>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

yaakovpraisler commented:
Hi Ulisse,

I wrote you on Slack DFIR workspace regarding to schedule us a call.
We can still do it today if you are available.
Please let me know.

Thanks

UlisseDallara commented:
Hi Yaakov,

I cannot see your message in Slack.

Can you send me your email address please and I will send a Teams invite?

Thanks,

Regards,

Ulisse
________________________________
From: Yaakov Praisler ***@***.***>
Sent: Thursday, February 16, 2023 9:05 AM
To: demisto/content ***@***.***>
Cc: Ulisse Dallara ***@***.***>; Mention ***@***.***>
Subject: Re: [demisto/content] Update CrowdStrikeFalcon.py (PR #24320)


Hi Ulisse,

I wrote you on Slack DFIR workspace regarding to schedule us a call.
We can still do it today if you are available.
Please let me know.

Thanks

—
Reply to this email directly, view it on GitHub<https://github.com/demisto/content/pull/24320#issuecomment-1432747997>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A5UZLHLI46EZX2Z7DGJ4P4LWXXUW7ANCNFSM6AAAAAAUQFP5SM>.
You are receiving this because you were mentioned.Message ID: ***@***.***>
